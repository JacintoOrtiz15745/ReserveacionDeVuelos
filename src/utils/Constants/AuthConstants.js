const GoogleId = '555459120478-5dvb5sj1o0996s86gg8bk69ogm7vdiu4.apps.googleusercontent.com';  
export const AuthConstants = {
  GoogleId,
};
