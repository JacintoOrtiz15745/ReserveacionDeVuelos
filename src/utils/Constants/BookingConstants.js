const TitleOrigin = 'Where are you now?';
const TitleDestination = 'Where will you be flying to?';
const TextInput = 'Select location';
const BtnText = 'Next';
const SelectDate = 'Select Date';
const Passengers = 'How many passengers?';
const BookingToRoute = 'BookingTo';
const BookingNowRoute = 'BookingNow';
const MyFlightsRoute = 'MyFlights';
const BookingDateRoute = 'BookingDate';
const BookingDoneRoute = 'BookingDone';
const BookingPassengersRoute = 'BookingPassenger';
const doneMessage = 'Your request was received.';
const finish = 'Finish';
const PickerData = [
  {
    label: 'Amsterdam, Netherlands',
    value: 'AMS,Netherlands',
  },
  {
    label: 'Berlin, Germany',
    value: 'BER,Germany',
  },
  {
    label: 'Paris, France',
    value: 'PAR,France',
  },
  {
    label: 'London, UK',
    value: 'LON,UK',
  },
  {
    label: 'Madrid, Spain',
    value: 'MAD,Spain',
  },
];
const passengerNumberData = [
  {
    id: '1',
    title: '1',
  },
  {
    id: '2',
    title: '2',
  },
  {
    id: '3',
    title: '3',
  },
  {
    id: '4',
    title: '4',
  },
  {
    id: '5',
    title: '5',
  },
];
export const BookingConstants = {
  doneMessage,
  TitleOrigin,
  TitleDestination,
  TextInput,
  BtnText,
  SelectDate,
  Passengers,
  BookingToRoute,
  BookingNowRoute,
  MyFlightsRoute,
  passengerNumberData,
  PickerData,
  BookingDateRoute,
  BookingPassengersRoute,
  BookingDoneRoute,
  finish,
};
