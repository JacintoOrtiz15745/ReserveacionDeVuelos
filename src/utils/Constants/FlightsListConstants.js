const planeIconName = 'plane';
const passengers = 'passengers';
const myFlightsTitle = 'My flights';
const flightDataExperimental = [
  {
    id: '1',
    originCountry: 'BEG',
    destinyCountry: 'AMS',
    originCity: 'Serbia',
    destinyCity: 'Netherlands',
    date: 'September 3, 2021',
    passengers: 5,
  },
  {
    id: '2',
    originCountry: 'BEG',
    destinyCountry: 'AMS',
    originCity: 'dsfsf',
    destinyCity: 'fdgdg',
    date: 'September 3, 2021',
    passengers: 5,
  },
  {
    id: '3',
    originCountry: 'BEG',
    destinyCountry: 'AMS',
    originCity: 'dsfsf',
    destinyCity: 'fdgdg',
    date: 'September 3, 2021',
    passengers: 5,
  },
  {
    id: '4',
    originCountry: 'BEG',
    destinyCountry: 'AMS',
    originCity: 'dsfsf',
    destinyCity: 'fdgdg',
    date: 'September 3, 2021',
    passengers: 5,
  },
  {
    id: '5',
    originCountry: 'BEG',
    destinyCountry: 'AMS',
    originCity: 'dsfsf',
    destinyCity: 'fdgdg',
    date: 'September 3, 2021',
    passengers: 5,
  },
  {
    id: '6',
    originCountry: 'BEG',
    destinyCountry: 'AMS',
    originCity: 'dsfsf',
    destinyCity: 'fdgdg',
    date: 'September 3, 2021',
    passengers: 5,
  },
  {
    id: '7',
    originCountry: 'BEG',
    destinyCountry: 'AMS',
    originCity: 'dsfsf',
    destinyCity: 'fdgdg',
    date: 'September 3, 2021',
    passengers: 5,
  },
];

export const FlightsListConstants = {
  planeIconName,
  passengers,
  myFlightsTitle,
  flightDataExperimental,
  
};
