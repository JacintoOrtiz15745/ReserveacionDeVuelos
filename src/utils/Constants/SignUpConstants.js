const SignUpTitle = 'Sign Up';
const NameInputTitle = 'First Name';
const EmailInputTitle = 'Email *';
const PasswordInputTitle = 'Password *';
const TermsCheckboxTitle = 'I agree to the Terms and Privacy Policy.';
const UpdatesCheckboxTitle = 'Subscribe for select product updates.';
const CharactersMessage =
  'Use 8 or more characters with a mix of letters, numbers, and simbols.';
const SignUpButton = 'Sign Up';
const GoogleSignUpButton = 'Sign Up with Google';
const HasAnAccount = 'Already have an account?';
const LogIn = 'Log In';
const nameErrorMessage = ' must be at least 3 characters.';
const emailErrorMessage = 'invalid Email format';
const passwordErrorMessage = 'must be at least 8 characters';
const OR = 'or';
const SignIn = 'Sign In';
const SignUp = 'Sign Up';
const noAccountMsg = 'DonÂ´t have an account?';
export const SignUpConstants = {
  SignUpTitle,
  NameInputTitle,
  EmailInputTitle,
  PasswordInputTitle,
  TermsCheckboxTitle,
  UpdatesCheckboxTitle,
  CharactersMessage,
  SignUpButton,
  GoogleSignUpButton,
  HasAnAccount,
  LogIn,
  nameErrorMessage,
  emailErrorMessage,
  passwordErrorMessage,
  OR,
  SignIn,
  noAccountMsg,
  SignUp,
};
