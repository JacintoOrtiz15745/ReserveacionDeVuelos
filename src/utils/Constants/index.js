
export * from './SignUpConstants';
export * from './FlightsListConstants'
export * from './AuthConstants'
export * from './BookingConstants'