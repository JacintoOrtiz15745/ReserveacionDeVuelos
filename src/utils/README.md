This folder will contain functions and utilities if needed.