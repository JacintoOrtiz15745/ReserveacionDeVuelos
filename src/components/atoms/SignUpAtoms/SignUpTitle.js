import React from 'react';
import {Text} from 'react-native';
import {styles} from '../../../styles/SignUpStyles';
import {SignUpConstants} from '../../../utils/Constants';
const SignUpTitle = () => {
  return <Text testID='signUpTitle' style={styles.title}>{SignUpConstants.SignUpTitle}</Text>;
};

export default SignUpTitle;
