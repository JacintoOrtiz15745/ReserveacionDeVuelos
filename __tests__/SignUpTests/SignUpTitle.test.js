import React from 'react';
import SignUpTitle from '../../src/components/atoms/SignUpAtoms/SignUpTitle';
import {shallow} from 'enzyme';
import { SignUpConstants } from "../../src/utils/Constants";
let wrapper;
beforeEach(() => {
  wrapper = shallow(<SignUpTitle />);
});
describe('SignUpTitle component renders as expected', () => {
  test('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('Title renders correclty', () => {
    const title = wrapper.findWhere(
      (node) => node.prop('testID') === 'signUpTitle',
    );
    expect(title.length).toBe(1);
  });
  test('Title string matches with our constant', () => {
    const title = wrapper.findWhere(
      (node) => node.prop('testID') === 'signUpTitle',
    );
    expect(title.props().children).toBe(SignUpConstants.SignUpTitle);
  });
});
