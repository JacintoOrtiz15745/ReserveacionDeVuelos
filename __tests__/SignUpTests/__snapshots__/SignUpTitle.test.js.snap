// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUpTitle component renders as expected renders correctly 1`] = `
<Text
  style={
    Object {
      "color": "#6472ee",
      "fontSize": 20,
      "fontWeight": "bold",
      "marginBottom": 10,
    }
  }
  testID="signUpTitle"
>
  Sign Up
</Text>
`;
